CI-API.Data/Repository/VolunteerMissionRepository.cs

```diff
@@ -0,0 +1,333 @@
+﻿using CI_API.Common.CommonMethods;
+using CI_API.Common.CommonModels;
+using CI_API.Core.CIDbContext;
+using CI_API.Core.Models;
+using CI_API.Core.ViewModel;
+using CI_API.Data.Interface;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CI_API.Data.Repository
+{
+    public class VolunteerMissionRepository : IVolunteerMissionRepository
+    {
+        #region Dependency Injection of DbContext 
+
+        private readonly CiPlatformDbContext _cIDbContext;
+
+        public VolunteerMissionRepository(CiPlatformDbContext cIDbContext)
+        {
+            _cIDbContext = cIDbContext;
+        }
+        #endregion
+
+        #region GetMissionData
+        public async Task<JsonResult> AllMissiondata(long missionId, long userId)
+        {
+            try
+            {
+                var isMissionId =await Task.FromResult(_cIDbContext.Missions.Where(m => m.MissionId == missionId).FirstOrDefault()) ;
+
+                if (isMissionId != null)
+                {
+                    var data1 = await Task.FromResult(from mission in _cIDbContext.Missions
```

Comment:
You already get mission record above use that here do not get again from db.
rename 'isMissionId ' to mission